app/backend/src/clj/teet/task/task_commands.clj

```diff
@@ -104,17 +104,124 @@
+                     :type :notification.type/task-waiting-for-review
+                     :project (project-db/task-project-id db task-id)})
+                  {})]
+               (concat (for [{id :db/id} (task-db/not-reviewed-task-files db user task-id)]
+                  {:db/id id
+                   :file/status :file.status/submitted})
+               (for [pid (task-db/not-reviewed-file-parts db task-id)]
+                 {:db/id (first pid)
+                  :file.part/status :file.part.status/waiting-for-review})))})
+
+(defcommand :task/task-part-review
+            {:doc "Submit task part for review and approve/reject."
+             :context {:keys [db user]}
+             :payload {task-id :task-id
+                       taskpart-id :taskpart-id
+                       result :result}
+             :project-id (project-db/task-project-id db task-id)
+             :authorization {:task/review {:eid task-id
+                                           :link :task/assignee}}
+             :transact
+             (case result
+               :accept (into [(merge {:db/id taskpart-id
+                                      :file.part/status :file.part.status/completed}
+                                     (task-db/task-part-completion-attributes user))
+                              (if-let [assignee (task-db/get-task-assignee-by-id db task-id)]
+                                (notification-db/notification-tx
+                                  db
+                                  {:from user
+                                   :to assignee
+                                   :target taskpart-id
+                                   :type :notification.type/task-part-review-accepted
+                                   :project (project-db/task-project-id db task-id)})
+                                {})]
+                             (for [{id :db/id} (task-db/task-part-file-listing db user taskpart-id)]
+                               {:db/id id
+                                :file/status :file.status/final}))
+               :reject (into [{:db/id taskpart-id
+                               :file.part/status :file.part.status/in-progress}
+                              (if-let [assignee (task-db/get-task-assignee-by-id db task-id)]
+                                (notification-db/notification-tx
+                                  db
+                                  {:from user
+                                   :to assignee
+                                   :target taskpart-id
+                                   :type :notification.type/task-part-review-rejected
+                                   :project (project-db/task-project-id db task-id)})
+                                {})]
+                             (for [{id :db/id} (task-db/task-part-file-listing db user taskpart-id)]
+                               {:db/id id
+                                :file/status :file.status/returned})))})
```

Comment:
Could these cases be split into their own functions?

Reply:
I would say it would make it a lot easier to read at least. Having functions `task-part-accept-tx` and `task-part-reject-tx`, and it would make the actual defcommand a lot shorter